plugins/hydra_ax_sweeper/tests/test_ax_sweeper_plugin.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import math
+import os
+from typing import Any, Callable, List, Optional
+
+import pytest
+import yaml
+from omegaconf import DictConfig
+
+from hydra.core.plugins import Plugins
+from hydra.plugins.sweeper import Sweeper
+
+# noinspection PyUnresolvedReferences
+from hydra.test_utils.test_utils import TSweepRunner  # noqa: F401
+from hydra.test_utils.test_utils import SweepTaskFunction
+from hydra_plugins.hydra_ax_sweeper import AxSweeper
+
+
+def test_discovery() -> None:
+    """
+    Tests that this plugin can be discovered via the plugins subsystem when looking for Sweeper
+    :return:
+    """
+    assert AxSweeper.__name__ in [x.__name__ for x in Plugins.discover(Sweeper)]
+
+
+class AxSweepTaskFunction(SweepTaskFunction):
+    """
+    Context function
+    """
+
+    def __enter__(self) -> "SweepTaskFunction":
+        super().__enter__()
+        print(self.temp_dir)
+        with open(f"{self.temp_dir}/optimization_results.yaml", "r") as f:
+            self.returns = yaml.safe_load(f)
+        return self
+
+
+@pytest.fixture(scope="function")  # type: ignore
+def ax_sweep_runner() -> Callable[
+    [
+        Optional[str],
+        Optional[str],
+        Optional[Callable[[DictConfig], Any]],
+        Optional[str],
+        Optional[List[str]],
+        Optional[bool],
+    ],
+    AxSweepTaskFunction,
+]:
+    def _(
+        calling_file: Optional[str],
+        calling_module: Optional[str],
+        calling_function: Optional[Callable[[DictConfig], Any]],
+        config_path: Optional[str],
+        overrides: Optional[List[str]],
+        strict: Optional[bool] = None,
+    ) -> AxSweepTaskFunction:
+        sweep = AxSweepTaskFunction()
+        sweep.calling_file = calling_file
+        sweep.calling_module = calling_module
+        sweep.calling_function = calling_function
+        sweep.config_path = config_path
+        sweep.strict = strict
+        sweep.overrides = overrides or []
+        return sweep
+
+    return _
+
+
+def banana(cfg: DictConfig) -> Any:
```

Comment:
this is the third function I see.
I was expecting two, one for example and one for test.